packages/network-controller/src/rpc-service/rpc-service.ts

```diff
@@ -483,20 +483,41 @@ export class RpcService implements AbstractRpcService {
     return await this.#policy.execute(async () => {
       const response = await this.#fetch(this.endpointUrl, fetchOptions);
 
-      if (response.status === 405) {
+      if (response.status === 401) {
+        // code: -33100
+        throw new JsonRpcError(-33100, 'Unauthorized.');
+      }
+
+      if (response.status === 405 || response.status === 501) {
+        // code: -32601
         throw rpcErrors.methodNotFound();
       }
 
       if (response.status === 429) {
+        // code: -32005
         throw rpcErrors.limitExceeded({
           message: 'Request is being rate limited.',
         });
       }
 
-      if (response.status === 503 || response.status === 504) {
-        throw rpcErrors.internal({
-          message:
-            'Gateway timeout. The request took too long to process. This can happen when querying logs over too wide a block range.',
+      if (response.status >= 400 && response.status < 500) {
+        // code: -32600
+        throw rpcErrors.invalidRequest();
+      }
+
+      if (
+        (response.status >= 500 && response.status < 600) ||
+        response.status === 402 ||
```

Comment:
This case will not be hit because the `response.status >= 400 && response.status < 500` condition comes first.

We can re-order the conditions so this comes before `response.status >= 400 && response.status < 500`

Reply:
Oops right

Reply:
Thanks fixed here https://github.com/MetaMask/core/pull/5807/commits/2614067ab63f73d5713edf4e232c4426e9427f2c